ddl/partition.go

```diff
@@ -112,6 +115,51 @@ func checkPartitionNameUnique(tbInfo *model.TableInfo, pi *model.PartitionInfo)
 	return nil
 }
 
+func checkPartitionFuncValid(expr ast.ExprNode) error {
+	switch v := expr.(type) {
+	case *ast.CaseExpr:
+		return ErrPartitionFunctionIsNotAllowed
+	case *ast.FuncCallExpr:
+		// check function which allowed in partitioning expressions
+		// https://dev.mysql.com/doc/mysql-partitioning-excerpt/5.7/en/partitioning-limitations-functions.html
+		switch v.FnName.L {
+		case ast.Abs, ast.Ceiling, ast.DateDiff, ast.Day, ast.DayOfMonth, ast.DayOfWeek, ast.DayOfYear, ast.Extract, ast.Floor,
+			ast.Hour, ast.MicroSecond, ast.Minute, ast.Mod, ast.Month, ast.Quarter, ast.Second, ast.TimeToSec, ast.ToDays,
+			ast.ToSeconds, ast.UnixTimestamp, ast.Weekday, ast.Year, ast.YearWeek:
+			return nil
+		default:
+			return ErrPartitionFunctionIsNotAllowed
+		}
+	case *ast.BinaryOperationExpr:
+		// The DIV operator is also supported; the / operator is not permitted.
+		// https://dev.mysql.com/doc/refman/5.7/en/partitioning-limitations.html
+		if v.Op == opcode.Div {
```

Comment:
since DIV is supported why we return error here?

Reply:
DIV operator is `opcode.IntDiv`, / operator is `opcode.Div`